data/trie/trieStorageManager.go

```diff
@@ -275,32 +275,44 @@ func (tsm *trieStorageManager) snapshot(msh marshal.Marshalizer, hsh hashing.Has
 		tr, err := newSnapshotTrie(tsm.db, msh, hsh, snapshot.rootHash)
 		if err != nil {
 			tsm.storageOperationMutex.Unlock()
+			isSnapshotsBufferEmpty, keys = tsm.isSnapshotsBufferEmpty()
```

Comment:
_, keys = ... ?

Reply:
No, every time the for loop breaks, `isSnapshotsBufferEmpty` must be recomputed, otherwise there will be a deadlock.

Reply:
But you have a break just after the next line, there would be no more loops

Reply:
You are right. Will replace the `break` with a `continue`.